libraries/AP_WindVane/AP_WindVane.h

```diff
@@ -0,0 +1,109 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Param/AP_Param.h>
+#include <AP_Math/AP_Math.h>
+
+class AP_WindVane
+{
+
+public:
+
+    enum WindVaneType {
+        WINDVANE_NONE       = 0,
+        WINDVANE_HOME_HEADING = 1,
+        WINDVANE_PWM_PIN    = 2,
+        WINDVANE_ANALOG_PIN = 3,
+        WINDVANE_SITL       = 10
+    };
+
+    AP_WindVane();
+
+    /* Do not allow copies */
+    AP_WindVane(const AP_WindVane &other) = delete;
+    AP_WindVane &operator=(const AP_WindVane&) = delete;
+
+    // destructor
+    ~AP_WindVane(void);
```

Comment:
We shouldn't ever be destructing this (or need a destructor).

Reply:
Ok, destructor removed

Reply:
..oops spoke too soon.  we need the delete in order to stop copies being made.  I'm pretty sure this is a device created by @peterbarker.

Reply:
You don't need it. There is a default destructor for all objects, declaring it here indicates you are actually going to try and put an implementation in that does something different.

Reply:
ok, removed!